src/roslibpy/ros.py

```diff
@@ -506,13 +506,23 @@ def get_nodes(self, callback, errback=None):
 
         service.call(ServiceRequest(), callback, errback)
 
+        if callback:
+            return
+
+        assert 'nodes' in result
+        return result['nodes']        
+
     def get_node_details(self, node, callback, errback=None):
         """Retrieve list subscribed topics, publishing topics and services of a specific node name."""
         service = Service(self, '/rosapi/node_details',
                           'rosapi/NodeDetails')
 
-        service.call(ServiceRequest({'node': node}), callback, errback)
-
+        result = service.call(ServiceRequest({'node': node}), callback, errback)
+        if callback:
+            return
+        
+        output= {'services': result['services'], 'subscribing':  result['subscribing'], 'publishing':result['publishing'] }
```

Comment:
This is a bit nitpicky, but I find the spacing in this line a little wonky. Moreover, I find the following easier to read:
`output = {attr: result.get(attr) for attr in ['services', 'subscribing', 'publishing']}`

Reply:
I just imitate how the other function is implemented. Mostly a quick fix for my project. Feel free to improve them :)